### PR TITLE
Added `w` property to Regridder and SpatialAverager

### LINKED ISSUE #275: Expose weights as a "normal" DataArray
The current weight matrix is flattened, with dimensions `(out_dim, in_dim)`. I think it would be useful, including for debugging, to reshape the weights so they're exposed as `lon_out, lat_out, lon_in, lat_in`. 

We could alternatively use the same dimension names as the input and output grids, but if they're both called lat and lon, this won't work.

### PR DESCRIPTION
**Pull request by huard**
Fixes #275

- Add `w` property
- Add test
- Also added a mark.fail test showing how averages on large regions can fail. Intended for later work.

### CODE REVIEW

**Comment by huard** [comment]
@charlesgauthier-udm 

Useful ?

File: xesmf/tests/test_frontend.py [hunk 1]
```diff
@@ -713,6 +736,27 @@ def test_spatial_averager(poly, exp):
     assert 'my_geom' in out.dims
 
 
+@pytest.mark.xfail
+def test_spatial_averager_with_zonal_region():
```
Comment (aulemahal) [inline xesmf/tests/test_frontend.py:740]: Should this be in this PR ? Looks related with your other work on the large regions issue.

**Review by aulemahal — APPROVED** [review]
Looks good. I guess this might be temporary, as @charlesgauthier-udm 's work might turn the logic around and make the "old" version the "extra" one and the 4D version the internal one.

**Comment by charlesgauthier-udm** [comment]
Could be useful yes. However, with more testing I find that the performance related to using 4D weights seems to be highly dependent on the input data shape and chunk sizes (with Dask) with the 2D weights sometimes outperforming the 4D weights.

**Comment by huard** [comment]
Let's not merge this then, and wait for Charles' results.

**Comment by charlesgauthier-udm** [comment]
I think we can merge it and I'll use the 4D w property to make another PR with the dask compatibility.

**Comment by huard** [comment]
Should I remove the non-related test ?

**Comment by aulemahal** [comment]
I think it's ok to keep it.

**Pull request merged**